### PR TITLE
chore: remove attrs from dd-trace-py

### PR DESCRIPTION
**Pull request by gnufede**
## Motivation
Remove `attrs` dependency

Merge queue dependencies:
- [x] https://github.com/DataDog/dd-trace-py/pull/9777
- [x] https://github.com/DataDog/dd-trace-py/pull/9860
- [x] https://github.com/DataDog/dd-trace-py/pull/9867

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by P403n1x87 — APPROVED** [review]
❤️

**Review by christophe-papazian — APPROVED** [review]

File: .riot/requirements/50d0377.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # This file is autogenerated by pip-compile with Python 3.11
 # by the following command:
 #
-#    pip-compile --no-annotate .riot/requirements/f20bd15.in
+#    pip-compile --no-annotate .riot/requirements/50d0377.in
```
Comment (brettlangdon) [inline .riot/requirements/50d0377.txt:5]: this is the only one that needed updating? 🤔 

there is also `min_compatible_versions.csv` and `lib-injection/min_compatible_versions.csv` which we might be able to remove it from as well?

Comment (gnufede) [reply .riot/requirements/50d0377.txt:5]: If I run the script `python ./scripts/min_compatible_versions.py`, other dependencies get updated in the file, is that OK?

Comment (brettlangdon) [reply .riot/requirements/50d0377.txt:5]: ah, yeah, makes sense!

as long as we run that and the "compile and prune lock files" one, then all good!

File: lib-injection/min_compatible_versions.csv [hunk 2]
```diff
@@ -62,18 +63,18 @@ elasticsearch8,~=8.0.1
 elasticsearch[async],0
 envier,==0.5.2
 exceptiongroup,0
-falcon,~=3.0
+falcon,~=3.0.0
 fastapi,~=0.64.0
 flask,~=0.12.0
 flask-caching,~=1.10.0
-flask-login,~=0.6.2
+flask-openapi3,0
```
Comment (gnufede) [inline lib-injection/min_compatible_versions.csv:70]: This one seems off

File: min_compatible_versions.csv [hunk 3]
```diff
@@ -62,18 +63,18 @@ elasticsearch8,~=8.0.1
 elasticsearch[async],0
 envier,==0.5.2
 exceptiongroup,0
-falcon,~=3.0
+falcon,~=3.0.0
 fastapi,~=0.64.0
 flask,~=0.12.0
 flask-caching,~=1.10.0
-flask-login,~=0.6.2
+flask-openapi3,0
```
Comment (gnufede) [inline min_compatible_versions.csv:70]: ditto

File: lib-injection/min_compatible_versions.csv (hunk 2 above)
Comment (brettlangdon) [reply lib-injection/min_compatible_versions.csv:70]: cc @emmettbutler

Comment (gnufede) [reply lib-injection/min_compatible_versions.csv:70]: https://pypi.org/project/flask-openapi3/#history

Comment (gnufede) [reply lib-injection/min_compatible_versions.csv:70]: Hmmm... I think this means we don't specify a min version, we just test with `latest` as per our riotfile

Comment (brettlangdon) [reply lib-injection/min_compatible_versions.csv:70]: oh! yeah, that was difficult to read, I didn't realize the name had `3` in it!

**Review by emmettbutler — APPROVED** [review]

**Pull request merged**